pkg/noderesourcetopology/score.go

```diff
@@ -135,7 +140,25 @@ func containerScopeScore(pod *v1.Pod, zones topologyv1alpha1.ZoneList, scorerFn
 	allocatablePerNUMA := createNUMANodeList(zones)
 
 	for i, container := range containers {
+		identifier := pod.Namespace + "/" + pod.Name + "/" + container.Name
 		contScore[i] = float64(scoreForEachNUMANode(container.Resources.Requests, allocatablePerNUMA, scorerFn, resourceToWeightMap))
+		klog.V(6).InfoS("container scope scoring", "container", identifier, "score", contScore[i])
+	}
+	finalScore := int64(stat.Mean(contScore, nil))
+	klog.V(4).InfoS("container scope scoring final node score", "finalScore", finalScore)
+	return finalScore, nil
+}
+
+func logNRT(desc string, nrtObj *topologyv1alpha1.NodeResourceTopology) error {
+	if !klog.V(6).Enabled() {
+		// avoid the expensive marshal operation
+		return nil
+	}
+
+	ntrJson, err := json.MarshalIndent(nrtObj, "", " ")
+	if err != nil {
+		return err
 	}
-	return int64(stat.Mean(contScore, nil)), nil
+	klog.V(6).Info(desc, " ", "noderesourcetopology=", string(ntrJson))
```

Comment:
nit: we can just write:
```json
klog.V(6).Info(desc, " noderesourcetopology=", string(ntrJson))``
```
I'm wondering if we should do MarshalIndent or we can just do Marshal. Logging in JSON format (and not just dumping the golang struct) is a good improvement. The compact JSON form, however, can be good enough because it's pretty fair to assume `jq` or any other JSON pretty printer is avaialble for the person checking the logs.
OTOH multi-line logs are unwieldly.

Reply:
> klog.V(6).Info(desc, " noderesourcetopology=", string(ntrJson))``

Definitely possible. I didn't do that, because I wanted the reviewers/readers to know that the space is explicit and not a typo.

> I'm wondering if we should do MarshalIndent or we can just do Marshal.

Just Marshling without indentation doesn't (literally!) look like a significant improvement over dumping the golang struct.

Reply:
fair point about doing this way to better inform the reader. I have mixed feelings about Marshal vs MarshalIndent, I'll have a deeper look later. I'm not really against unsing MarshalIndent though